doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst

```diff
@@ -0,0 +1,300 @@
+Key slots that are tagged with CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED also have to be decrypted to a temporary push location in RAM before they are used by CRACEN.
+
+When the application is built with TF-M, this temporary push location is protected inside the secure processing environment to avoid exposing the key material to the non-secure application.
+If TF-M is not used, the keys are pushed to a reserved RAM area at location 0x20000000-0x20000064 (kmu_push_area).
+
+The following special error codes are available:
+
+* PSA_ERROR_INVALID_HANDLE: Attempting an operation is attempted on an empty KMU slot
+* PSA_ERROR_NOT_PERMITTED: Trying an operation on a revoked key
+* PSA_ERROR_HARDWARE_FAILURE: Key slot has invalid data
+* PSA_ERROR_CORRUPTION_DETECTED: Key slot has invalid data
+
+CRACEN IKG
+**********
+
+The Isolated Key Generator (IKG) can derive 3 keys from a 384-bit seed value.
+This seed value can be stored in the KMU, and pushed directly to the IKG without being accessible by the CPU.
+The derived keys are also not accessible by the CPU, but they can be used for cryptographic operations by CRACEN.
+
+These keys are also accessed using normal PSA Crypto APIs, and are referenced by special built-in key IDs.
+
++-----------------+---------------------------------+---------------------------------------------------+
+| Key type        | Key ID                          | Description                                       |
++=================+=================================+===================================================+
+| ECC secp256r1   | CRACEN_BUILTIN_IDENTITY_KEY_ID  | Used for signing/verification.                    |
++-----------------+---------------------------------+---------------------------------------------------+
+| AES 256-bit     | CRACEN_BUILTIN_MKEK_ID          | Used for encryption/decryption or key derivation. |
++-----------------+---------------------------------+---------------------------------------------------+
+| AES 256-bit     | CRACEN_BUILTIN_MEXT_ID          | Used for encryption/decryption or key derivation. |
++-----------------+---------------------------------+---------------------------------------------------+
+
+The keys are not exportable, except for the public key associated with the asymmetric key.
+
+Configuration
+=============
+
+To enable KMU support, do the following:
+
+.. code-block:: console
+
+   CONFIG_NRF_SECURITY=y
+   CONFIG_MBEDTLS_PSA_CRYPTO_C=y
+   CONFIG_PSA_CRYPTO_DRIVER_CRACEN=y
+
+Enabling support for encrypted key scheme (CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED), should look as follows:
+
+.. code-block:: console
+
+   CONFIG_PSA_WANT_ALG_GCM=y
+   CONFIG_PSA_WANT_KEY_TYPE_AES=y
+   CONFIG_PSA_WANT_AES_KEY_SIZE_256=y
+   CONFIG_PSA_WANT_ALG_SP800_108_COUNTER_CMAC=y
```

Comment:
I checked, we need these two configs as well to enable KMU support

CONFIG_PSA_WANT_ALG_CMAC=y
CONFIG_PSA_WANT_ALG_ECB_NO_PADDING=y

Reply:
At which point should we we have some umbrella config so users don't have to add all these to "clutter" their project config to use the KMU?

Reply:
I agree, would be nice to have one config to enable KMU support. Wonder if this is the right place to add a task for this https://nordicsemi.atlassian.net/browse/NCSDK-7747